store/src/directory.rs

```diff
@@ -392,21 +812,80 @@ where
+                        log::trace!(
+                            "Error opening metadata file {}: {:?}",
+                            path_meta.display(),
+                            e
+                        );
+                        continue;
+                    }
+                };
+                let metadata: FdoMetadata = match serde_cbor::from_reader(&file_meta) {
+                    Ok(data) => data,
+                    Err(e) => {
+                        return Err(StoreError::Unspecified(format!(
+                            "Error reading metadata from '{}': {:?}",
+                            path_meta.display(),
+                            e
+                        )))
+                    }
+                };
+                let val = metadata.map.get(crate::MetadataKey::<MKT>::Ttl.to_key());
+                match val {
+                    Some(ttl) => {
+                        if SystemTime::now() < ttl_from_disk(&ttl)? {
+                            continue;
+                        }
+                    }
+                    None => {
+                        continue;
+                    }
+                };
+                log::trace!("File at {} has expired, attempting removal", path.display());
+                if let Err(e) = fs::remove_file(&path) {
+                    log::warn!("Error deleting expired file {}: {}", path.display(), e);
+                }
+                if let Err(e) = fs::remove_file(&path_meta) {
+                    log::warn!(
+                        "Error deleting expired metadata file {}: {}",
+                        path_meta.display(),
+                        e
+                    );
+                }
+            } else {
+                let ttl = match xattr::get(
+                    &path,
+                    format_xattr(crate::MetadataKey::<MKT>::Ttl.to_key()),
+                ) {
+                    Err(e) => {
+                        log::trace!("Error looking up TTL xattr for {}: {:?}", path.display(), e);
+                        continue;
+                    }
+                    Ok(None) => continue,
+                    Ok(Some(val)) => ttl_from_disk(&val)?,
+                };
+                if SystemTime::now() < ttl {
                     continue;
                 }
-                Ok(None) => continue,
-                Ok(Some(val)) => ttl_from_disk(&val)?,
-            };
-            if SystemTime::now() < ttl {
-                continue;
-            }
-            log::trace!("File at {} has expired, attempting removal", path.display());
-            if let Err(e) = fs::remove_file(&path) {
-                log::info!("Error deleting expired file {}: {}", path.display(), e);
+                log::trace!("File at {} has expired, attempting removal", path.display());
+                if let Err(e) = fs::remove_file(&path) {
+                    log::info!("Error deleting expired file {}: {}", path.display(), e);
+                }
```

Comment:
this remove_file is basically duplicate with the other in !xattr_enabled so just move it out of the if/else at the end